File: threedollar-push/src/main/java/com/depromeet/threedollar/push/common/dto/response/ApiResponse.java
package com.depromeet.threedollar.push.common.dto.response;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

@ToString
@Getter
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class ApiResponse<T> {

    public static final ApiResponse<String> OK = success("OK");

    private String resultCode;

    private String message;

    private T data;

    public static <T> ApiResponse<T> success(T data) {
        return new ApiResponse<>("", "", data);
    }

}

File: threedollar-consumer/src/main/kotlin/com/depromeet/threedollar/consumer/controller/ApiResponse.kt
package com.depromeet.threedollar.consumer.controller

class ApiResponse<T>(
    val code: String,
    val message: String,
    val data: T?
) {
    companion object {
        val SUCCESS = success("OK")
        fun <T> success(data: T): ApiResponse<T> {
            return ApiResponse<T>("", "", data)
        }
    }

}

